galgebra/ga.py

```diff
@@ -700,7 +703,37 @@ def _build_basis_blade_symbol(self, base_index):
                 raise ValueError('!!!!No unique root symbol for wedge_print = False!!!!')
         return Symbol(symbol_str, commutative=False)
 
-    def _build_bases(self):
+    def build_cobases(self, coindexes=None):
```

Comment:
xref https://github.com/pygae/galgebra/issues/215.

I think I understand the rough intent here, although it's not clear to me why a separate set of indices or signs are needed.

Reply:
There is a "canonical" way for building cobases (see John Browne's book) and PGA uses some kind of "tricky" way for building it (J and Jinv stuff). My implementation is probably not the best we can do.

Reply:
@enkimute, is "cobase" consistent with the terminology you'd use?

Reply:
The sign is for using the orthogonal wedge implementation which needs a canonical basis. PGA basis isn't canonical too.

Reply:
When you say the PGA basis isn't canonical, do you mean that the default is not to use lexicographically-order basis blades? Eg, `e132` is considered a "base" blade in that algebra, rather than `e123`?

Reply:
Yeap

Reply:
This forum post captures the issue I assume?

https://discourse.bivector.net/t/dual-operator-disambiguation/59/45

Reply:
Yeap

Reply:
Another way to handle this might just be in the printing only

Reply:
It would be very nice if you are able to run test_pga.py without adding the custom indexes and signs. I don't see how we can achieve this.